misk/src/test/kotlin/misk/logging/LoggingTest.kt

```diff
@@ -1,13 +1,109 @@
 package misk.logging
 
+import ch.qos.logback.classic.Level
+import ch.qos.logback.classic.LoggerContext
+import com.google.inject.util.Modules
+import misk.MiskServiceModule
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
 import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.containsExactly
 import org.junit.jupiter.api.Test
+import org.slf4j.LoggerFactory
+import org.slf4j.MDC
+import javax.inject.Inject
 
+@MiskTest(startService = true)
 class LoggingTest {
+  @MiskTestModule
+  val testModule = Modules.combine(MiskServiceModule(), LogCollectorModule())
+
+  @Inject private lateinit var logCollector: LogCollector
+
   private val logger = getLogger<LoggingTest>()
+  private val loggerContext = LoggerFactory.getILoggerFactory() as LoggerContext
 
   @Test
   fun loggerNameIsBasedOnTheOuterClass() {
     assertThat(logger.name).isEqualTo("misk.logging.LoggingTest")
   }
+
+  @Test
+  fun loggingWithTags() {
+    // clear existing messages
+    logCollector.takeEvents(LoggingTest::class)
+
+    // Log without tags
+    logger.error(IllegalStateException("failed!")) { "untagged error" }
+    logger.info { "untagged info" }
+
+    val untaggedEvents = logCollector.takeEvents(LoggingTest::class)
+    assertThat(untaggedEvents).hasSize(2)
+    assertThat(untaggedEvents[0]).satisfies {
+      assertThat(it.level).isEqualTo(Level.ERROR)
+      assertThat(it.message).isEqualTo("untagged error")
+      assertThat(it.throwableProxy.className).isEqualTo(IllegalStateException::class.qualifiedName)
+      assertThat(it.mdcPropertyMap).isEmpty()
+    }
+    assertThat(untaggedEvents[1]).satisfies {
+      assertThat(it.level).isEqualTo(Level.INFO)
+      assertThat(it.message).isEqualTo("untagged info")
+      assertThat(it.throwableProxy).isNull()
+      assertThat(it.mdcPropertyMap).isEmpty()
+    }
+
+    // Log with tags
+    logger.info("user-id" to "blerb", "alias-id" to "d6F1EF53") { "tagged info" }
```

Comment:
TIL https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/to.html, interesting syntax

Reply:
nice! this is super easy to use